### ISSUE TITLE
Add a method for finding strictly 1 entity

### ISSUE THREAD

**Issue by wasd171** [mikro-orm/mikro-orm#3051]
**Is your feature request related to a problem? Please describe.**
Often when we query the database the condition that we would like to express is "does there exist a single entity that fits these criteria". Currently `EntityManager` has a `.findOneOrFail` method which solves half of the problem - it will throw in case there are no entities but will happily return the first one in case there are multiple, meaning that real name of the method is probably closer to `.findFirstOrFail`.

**Describe the solution you'd like**
I would like to be able to use `em.findSingleOrFail` or even `em.single` method. Signature of the method should be the same as for the current `.findOneOrFail`. This method should throw in case there are 0 or 2+ entities instead of 1.

**Describe alternatives you've considered**
We have 2 major options:
1. Always use `.find(..., {limit: 2})` and check for `length === 0 || length > 1` in an external helper (this is our current approach)
2. Monkey-patch `em` with the method that we need that will call `.find` internally

**Additional context**
I think that documentation from similar C# methods can be useful to consider:
[Enumerable.Single](https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.single?view=net-6.0)
[Enumerable.First](https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.first?view=net-6.0)

**Comment by B4nan** [comment]
I don't think we need a separate method, rather it should be an option for `findOneOrFail` that can be controlled both locally and globally (like the custom fail handler).

PR welcome

**Comment by d3v0ps** [comment]
> I don't think we need a separate method, rather it should be an option for `findOneOrFail` that can be controlled both locally and globally (like the custom fail handler).
> 
> PR welcome

As the findOneOrFail internally calls the `findOne` method, should we add the option to the `findOne` instead? In that case, what should we expect the findOne to have the same behavior?

**Comment by B4nan** [comment]
i guess it could call em.find conditionally, no need for changing the driver interface.

**Comment by B4nan** [comment]
#3088 adds `strict` option to the `findOneOrFail` method
